internal/controllers/synthesis/exec_io_test.go

```diff
@@ -0,0 +1,81 @@
+package synthesis
+
+import (
+	"testing"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildPodInput(t *testing.T) {
```

Comment:
Nice unit tests 🥇 

Can you also add a TODO for an integration test to cover inputs? Not worth adding now, but probably a good idea longer term